Fixes for disk resize

cfergeau opened the pull request:
Most important one is `Resize disk images at creation time as well` which is required for
`crc delete && crc start --disk-size xxx` to work.

cfergeau committed:
Add check for shrinking when resizing disk images

libvirt will report an error when this happens, but we can check
for it before even calling into libvirt.

cfergeau committed:
Resize disk images at creation time as well

Resizing disk images only works after initial creation of the VM.
It's a perferctly valid use case to resize the disk of the VM at
creation time, so the machine driver should support this.
This commit achieves this by reusing the same code as the one used by
UpdateConfigRaw

jsliacan commented on pkg/libvirt/libvirt.go:
```diff
@@ -162,6 +162,26 @@ func (d *Driver) setVcpus(cpus uint) error {
+		return err
+	}
+
+	if newCapacity == 0 || newCapacity == diskCapacity {
```
The second check (`newCapacity == diskCapacity`) is the same as L155 in pkg/libvirt/storage.go given that `capacity` and `diskCapacity` both denote the return value of `d.getVolCapacity()`.

gbraad replied:
This can be a check by itself in a separate function, preventing the name maybe here;

checkIfResizeNeeded(capacity) -> bool, err (needed, failureError)
resizeDiskImage(capacity) -> bool, err  (success, failureError)

cfergeau replied:
Fixed.

gbraad commented on pkg/libvirt/libvirt.go:
```diff
@@ -162,6 +162,26 @@ func (d *Driver) setVcpus(cpus uint) error {
 	return nil
 }
 
+func (d *Driver) maybeResizeDiskImage(newCapacity uint64) error {
```
maybe sound indecisive.

resizeIfNeeded makes more sense

gbraad reviewed (changes requested):
Using indecisive function names might lead to confusion; The intent of the function is not clear. When would it decide a maybe? Avoid anything else than yes or no.

cfergeau committed:
Silence disk resizing debug

log.Debugf is enough for debug output when it's useful to have,
log.Infof is not needed.
